client/src/components/ActiveChat/ActiveChat.js

```diff
@@ -1,30 +1,48 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { makeStyles } from "@material-ui/core/styles";
 import { Box } from "@material-ui/core";
 import { Input, Header, Messages } from "./index";
 import { connect } from "react-redux";
+import { readMessage } from "../../store/utils/thunkCreators";
 
 const useStyles = makeStyles(() => ({
   root: {
     display: "flex",
     flexGrow: 8,
-    flexDirection: "column"
+    flexDirection: "column",
   },
   chatContainer: {
     marginLeft: 41,
     marginRight: 41,
     display: "flex",
     flexDirection: "column",
     flexGrow: 1,
-    justifyContent: "space-between"
-  }
+    justifyContent: "space-between",
+  },
 }));
 
 const ActiveChat = (props) => {
   const classes = useStyles();
   const { user } = props;
   const conversation = props.conversation || {};
 
+  // Set messages in conversation isRead to be true if there are any unread messages
+  const setUnreadMessagesToBeRead = async () => {
+    if (!conversation || !conversation.id) return;
+
+    for (let message of conversation.messages) {
+      if (message.isRead == false) {
+        // Set conversation to be read as long as at least 1 message is unread
+        await props.readMessage(conversation.otherUser.id, conversation.id);
+        return;
+      }
+    }
+  };
+
+  useEffect(() => {
+    setUnreadMessagesToBeRead();
+  }, [conversation]);
+
```

Comment:
This works, but there are likely more efficient ways to do this.

Reply:
i tried to improve by setting to read when it is clicked to active state